### DISCUSSION TITLE
UI overhaul

### DISCUSSION THREAD

**Discussion by HathHub** [SmartlyDressedGames/Unturned-3.x-Community#3745] [Ideas]
We would like to request prioritization of a new UI system for Unturned. This update would greatly improve the quality of life for developers and offer server owners the opportunity to acquire more engaging gameplay content. We recommend utilizing React to create UI components that can be easily displayed, enabling in-time creation of interfaces and significantly speeding up UI plugin development. All of this should have a respective callback system to be able to receive button clicks and such.

This is one of the few remaining systems desperate for an update, main purpose is to have a better overall and enganing experience which would open many doors for the community.

We appreciate any additional suggestions and key factors that can support the approval of this overhaul. Thank you for your time and consideration.

**Comment by Corbyncc** [comment]
Nelson, a UI overhaul is desperately needed. I understand that you may feel its low priority, and while what we have right now is bearable, its pretty limiting and can be frustrating to work with at times. I would also like to see something like React framework implemented (prehaps [THIS](https://github.com/ReactUnity/core) library) or for you to implement an API for Glazier UI on the server. I am personally more partial towards a solution like React as it would allow basically unlimited freedom, and all you would have to do is implement some sort of way for us to fire callbacks to/from the server. FiveM has had a lot of success implementing UI this way. But as plugin developers we would be happy to have literally ANY supported system.

**Comment by blazethrower320** [comment]
Yes. I agree.

**Comment by ZafraFam** [comment]
Please ;-;

**Comment by sunnamed434** [comment]
Magic idea

**Comment by realtrollman2319** [comment]
I agree too

**Comment by liebesleid-eth** [comment]
+1

**Comment by educatalan02** [comment]
+1

**Comment by DanielWillett** [comment]
+1

**Comment by XanderCodes** [comment]
+1

**Comment by benvalkin** [comment]
+1

**Comment by PARTOVIY** [comment]
+1

**Comment by Jdance-Media** [comment]
I agree, but excessively requesting it isn't going to change things 🤣. At some point we will get it, after all it is planned on the Trello page.

**Comment by MolyiDev** [comment]
+1

**Comment by rube200** [comment]
it would be good

**Comment by Corbyncc** [comment]
ROLLED AND IGNORED BY LULSON

**Comment by liebesleid-eth** [comment]
T-T
